postgres-types/src/lib.rs

```diff
@@ -769,6 +769,40 @@ pub trait ToSql: fmt::Debug {
         ty: &Type,
         out: &mut BytesMut,
     ) -> Result<IsNull, Box<dyn Error + Sync + Send>>;
+
+    /// Specify the encode format
+    fn encode_format(&self) -> i16 { 1 }
+
+    /// return string representation
+    fn as_string(&self) -> String {
+        panic!("as_string not implemented for {:?}", self)
+    }
+}
+
+
+/// A Wrapper type used for sending query parameters encoded as unknown.
+#[derive(Debug)]
+pub struct Unknown<'a>(pub &'a (dyn ToSql + Sync));
```

Comment:
What's the point of using `ToSql` in here if the only method being used is `as_string`? It seems like just using `&str` would suffice.

Reply:
The name `Unknown` doesn't seem right to me. There is no "unknown" encoding in the Postgres protocol, just binary and text.

Reply:
> What's the point of using ToSql ...

You are right, it should be just `pub struct Unknown<'a>(pub &'a str);` (which would also eliminate the need for `as_string` in the `ToSql` trait)
I arrived at this while thinking "i have a `Vec<&'a (dyn ToSql + Sync)>` and i want to map a function over it and wrap each value in `Unknown` so that it's encoded as string" but that is not the best way as you pointed out, that initial vector in my case is anyway conceptually a `Vec<String>` anyway so your observation is correct.

> The name Unknown doesn't seem right to me...

The name here signifies not the way we are sending the value but the (pseudo) "type" of the value we are sending, just the way we are conceptually sending over `int`/`text`/... we are sending in this case a `unknown` variable type that only incidentally needs to be encoded in text mode.